### ISSUE TITLE
Themevariant expand-last doens't work properly on component resize

### ISSUE THREAD

**Issue by eforest** [vaadin-component-factory/anchor-nav#4]
Themevariant expand-last doens't work properly on component resize.

CSS variable --_expand-last-height is calculated only on slotchange, but not on resize.
This leads to a situation where scrolling down to last section won't highlight it's according last tab.

Steps to reproduce with demo: https://incubator.app.fi/anchor-nav-demo/anchor-nav
* open demo with small browser window height
* resize browser window to large(r) height / full screen
* scroll down
* => last tab will not be highlighted

Can be caused not only by browser window resize, but by any change in size (some other component is expanded for example).

**Comment by jcgueriaud1** [comment]
Hi,

It should be fixed in v1.1.1. Can you try it and reopen it if it's not the case?

**Comment by eforest** [comment]
Hi @jcgueriaud1 , 
I tried to update, but got problems on webpack. 
(version isn't listed here as well: https://vaadin.com/directory/component/vaadin-component-factoryvcf-anchor-nav)
```
No matching version found for @vaadin-component-factory/vcf-anchor-nav@1.2.10.
```

**Comment by jcgueriaud1** [comment]
It's in npm: https://www.npmjs.com/package/@vaadin-component-factory/vcf-anchor-nav
I also updated the webcomponent in the Vaadin directory but it shouldn't affect your build, since the build is using npm.

And github can deploy it. Can you try to clean your node_modules? (maybe there was an issue with npm during the process)

**Comment by eforest** [comment]
Now it worked! 
But I found another problem. There's now an error when resizing. Unfortunately stacktrace doesn't tell very much.

````
ResizeObserver loop completed with undelivered notifications. (:0)
nD @ client-4C23589935A74206212F474A719FC41C.cache.js:213
ao @ client-4C23589935A74206212F474A719FC41C.cache.js:1016
_n @ client-4C23589935A74206212F474A719FC41C.cache.js:787
bo @ client-4C23589935A74206212F474A719FC41C.cache.js:607
lj @ client-4C23589935A74206212F474A719FC41C.cache.js:1037
Ab @ client-4C23589935A74206212F474A719FC41C.cache.js:795
h @ client-4C23589935A74206212F474A719FC41C.cache.js:963
a.onerror @ client-4C23589935A74206212F474A719FC41C.cache.js:964
V @ vaadin-bundle-efecc7eb00ddb45a1023.cache.js:4917
i @ vaadin-bundle-efecc7eb00ddb45a1023.cache.js:4917
error (async)
```

**Comment by jcgueriaud1** [comment]
Are you able to reproduce it in this demo? https://incubator.app.fi/anchor-nav-demo/anchor-nav#section-4

I resized the component and it looks ok.
Can you you give me more information about the browser, Vaadin version,... ?

**Comment by eforest** [comment]
Hm, can't reproduce it there. 
Using Vaadin 14.6.4 / Chrome 91 / windows.

**Comment by jcgueriaud1** [comment]
Can you try to create or extract the code to reproduce the issue?

**Comment by eforest** [comment]
Here we go. With this small sample I can reproduce it locally (Vaadin 14.6.4 / Chrome 91 / windows)

```
@Route("anchornav")
@RouteAlias("")
public class DemoAnchorNav extends Div {

    private static final long serialVersionUID = 1L;

    public DemoAnchorNav() {
        dtAnchorNav();
    }

    private void dtAnchorNav() {
        AnchorNav nav = new AnchorNav();
        nav.addThemeVariants(AnchorNavVariant.EXPAND_LAST);

        nav.addSection("mytitle1", content());
        nav.addSection("mytitle2", content());
        nav.addSection("mytitle3", content());
        nav.addSection("mytitle4", content());
        nav.addSection("mytitle5", content());
        nav.addSection("mytitle6", content());
        nav.addSection("mytitle7", content());
        nav.addSection("mytitle8", content());
        nav.addSection("mytitle9", content());
        add(nav);
    }

    private VerticalLayout content() {
        Paragraph p1 = new Paragraph(
            "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. ");
        Paragraph p2 = new Paragraph(
            "Duis autem vel eum iriure dolor in hendrerit in vulputate velit esse molestie consequat, vel illum dolore eu feugiat nulla facilisis at vero eros et accumsan et iusto odio dignissim qui blandit praesent luptatum zzril delenit augue duis dolore te feugait nulla facilisi. Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat.");
        VerticalLayout layout = new VerticalLayout(p1, p2);
        return layout;
    }
}
```

**Comment by eforest** [comment]
Hi @jcgueriaud1, any news here?

**Comment by jcgueriaud1** [comment]
Ok. I can reproduce the issue. Thanks for the sample. I'll check what I can do.

**Comment by jcgueriaud1** [comment]
Fixed and published in version 1.1.2.
https://vaadin.com/directory/component/anchor-nav-for-flow

**Comment by eforest** [comment]
Hi @jcgueriaud1, I could verify this works fine now. Except when changing only browser height, then there's no recalculation taking place and we're back at the original problem.

Can be reproduced with code from above:
* open view in small browser window
* raise browser height (only height, not width)
* try to scroll down to last section
* => last tab is not highlighted

**Comment by eforest** [comment]
Hi @jcgueriaud1, any news on this?

**Comment by jcgueriaud1** [comment]
Hi,

Unfortunately I don't have time to change this. So it's still pending

**Comment by jcgueriaud1** [comment]
Closed by #10
